Formula/glib-utils.rb

```diff
@@ -0,0 +1,105 @@
+class GlibUtils < Formula
+  include Language::Python::Shebang
+
+  desc "Python utilities for GLib"
+  homepage "https://developer.gnome.org/glib/"
+  url "https://download.gnome.org/sources/glib/2.72/glib-2.72.2.tar.xz"
+  sha256 "78d599a133dba7fe2036dfa8db8fb6131ab9642783fc9578b07a20995252d2de"
+  license "LGPL-2.1-or-later"
+
+  depends_on "meson" => :build
+  depends_on "ninja" => :build
+  depends_on "pkg-config" => :build
+  depends_on "glib"
+  depends_on "python@3.10"
+
+  def install
+    # TODO: This is a workaround for `brew audit --new-formula`.
+    #       Use `patch` rather than `inreplace` (see also `glib`).
+    # replace several hardcoded paths with homebrew counterparts
+    inreplace "gio/xdgmime/xdgmime.c",
+              'xdg_data_dirs = "/usr/local/share/:/usr/share/";',
+              "xdg_data_dirs = \"#{HOMEBREW_PREFIX}/share/:/usr/local/share/:/usr/share/\";"
+    inreplace "glib/gutils.c",
+              'data_dirs = "/usr/local/share/:/usr/share/";',
+              "data_dirs = \"#{HOMEBREW_PREFIX}/share/:/usr/local/share/:/usr/share/\";"
+
+    # Point the headers and libraries to `glib`.
+    # The headers and libraries will be removed later because they are provided by `glib`.
+    glib = Formula["glib"]
+    args = std_meson_args.delete_if do |arg|
+      arg.start_with?("--prefix=", "--includedir=", "--libdir=")
+    end
+    args += %W[
+      --prefix=#{prefix}
+      --includedir=#{glib.opt_include}
+      --libdir=#{glib.opt_lib}
+    ]
+
+    # Disable dtrace; see https://trac.macports.org/ticket/30413
+    # and https://gitlab.gnome.org/GNOME/glib/-/issues/653
+    args += %W[
+      --default-library=both
+      --localstatedir=#{var}
+      -Diconv=auto
+      -Dgio_module_dir=#{HOMEBREW_PREFIX}/lib/gio/modules
+      -Dbsymbolic_functions=false
+      -Ddtrace=false
+    ]
+
+    mkdir "build" do
+      system "meson", *args, ".."
+      system "ninja", "-v"
+
+      # Skip files already in glib
+      Formula["meson"].opt_libexec.cd do
+        system "bin/python3", "-c", pyscript
+      end
+      system "ninja", "install", "-v"
+    end
+
+    # Delete non python files because they are provided by `glib`
+    dummy_python_shebang_rewrite_info = python_shebang_rewrite_info("#{HOMEBREW_PREFIX}/bin/python3")
+    Dir.glob(prefix/"**/*").each do |path|
```

Comment:
We can probably save this for next time, but each instance of the above can be replaced with.
```suggestion
    prefix.glob("**/*").each do |path|
```

Reply:
> Also, why not `prefix.find`?

It sounds better, I'm quite familiar with the Ruby Standard Libraries though. The same changes here:

```diff
  # Delete empty directories
- Dir.glob(prefix/"**/").reverse_each { |d| Dir.rmdir(d) if Dir.entries(d).size == 2 }
+ prefix.glob(/"**/").reverse_each { |d| d.rmdir if d.entries.size == 2 }
```

I can push a new commit if the CI permits.

Reply:
Unfortunately pushing a new commit would require us to rerun CI again in order to be able to publish the bottles AFAIK.  So this should be handled in a follow up PR with dependent testing disabled, which will not require the `CI-long-timeout` label.